### PR TITLE
refactors gun default attachment code

### PR DESCRIPTION
**Pull request by spookydonut**
ignoring `starting_attachment_types` and messing with flags in gun Inits is hacky and bad

### CODE REVIEW

**Comment by Ghommie** [comment]
the `var/list/starting_attachment_types` in the root comes with a dumb comment indicating it should only be just used for attachments that can be effectively removed. So if you want, go ahead and "removable" string from it.

**Comment by spookydonut** [comment]
Only because most attachments are removable by default and no one added subtypes properly

File: code/_onclick/hud/screen_objects.dm [hunk 1]
```diff
@@ -631,7 +631,7 @@
 
 	var/obj/item/weapon/gun/G = user.get_active_held_item()
 
-	if(!G || !G.has_ammo_counter() || !G.hud_enabled)
+	if(!G || !(G.flags_gun_features & GUN_AMMO_COUNTER) || !G.hud_enabled)
```
Comment (Rohesie) [inline code/_onclick/hud/screen_objects.dm]:
```suggestion
	if(!G?.hud_enabled || !(G.flags_gun_features & GUN_AMMO_COUNTER))
```

File: code/_onclick/hud/screen_objects.dm [hunk 2]
```diff
@@ -645,7 +645,7 @@
 
 	var/obj/item/weapon/gun/G = user.get_active_held_item()
 
-	if(!G || !istype(G) || !G.has_ammo_counter() || !G.hud_enabled || !G.get_ammo_type() || isnull(G.get_ammo_count()))
+	if(!G || !istype(G) || !(G.flags_gun_features & GUN_AMMO_COUNTER) || !G.hud_enabled || !G.get_ammo_type() || isnull(G.get_ammo_count()))
```
Comment (Rohesie) [inline code/_onclick/hud/screen_objects.dm]:
```suggestion
	if(!istype(G) || !(G.flags_gun_features & GUN_AMMO_COUNTER) || !G.hud_enabled || !G.get_ammo_type() || isnull(G.get_ammo_count()))
```

File: code/__DEFINES/conflict.dm [hunk 3]
```diff
@@ -35,22 +35,23 @@
...
+#define GUN_AMMO_COUNTER		(1 << 7)
+#define GUN_BURST_ON			(1 << 8)
+#define GUN_BURST_FIRING		(1 << 9)
+#define GUN_FLASHLIGHT_ON		(1 << 10)
+#define GUN_WIELDED_FIRING_ONLY	(1 << 11)
+#define GUN_HAS_FULL_AUTO		(1 << 12)
+#define GUN_FULL_AUTO_ON		(1 << 13)
+#define GUN_POLICE				(1 << 14)
+#define GUN_ENERGY				(1 << 15)
+#define GUN_LOAD_INTO_CHAMBER	(1 << 16)
```
Comment (DominikPanic) [inline code/__DEFINES/conflict.dm]: You should also add this into the appropriate bitfield so us admins can ~~bus~~ VV things easily.

**Comment by DominikPanic** [comment]
SMG's with this PR cannot burstfire.

**Pull request merged**